### PR TITLE
chore(stylelint): fix `Button`, `ComboButton`, and some `DataDecorator` warnings

### PR DESCRIPTION
**Pull request by SimonFinney**
### Affected issue

Contributes to #973 

### Proposed changes

- Fix `Button`, `ComboButton`, and some `DataDecorator` warnings
- Update global motion variables to point to Carbon motion tokens

### CODE REVIEW

File: src/__tests__/scss/__snapshots__/styles-test.js.snap [hunk 1]
```diff
@@ -3594,7 +3594,7 @@ body {
 }
 
 .security--card__link {
-  transition-duration: 0.2s;
+  transition-duration: 240ms;
```
Comment (areddon) [inline src/__tests__/scss/__snapshots__/styles-test.js.snap:3597]: Are there no tokens for the durations?

Comment (areddon) [reply src/__tests__/scss/__snapshots__/styles-test.js.snap:3597]: > $transition-duration: 0.2s;
$transition-duration: $moderate-02;

**Review by areddon — CHANGES_REQUESTED** [review]
A few `transition-duration`s not using the Carbon tokens.

File: src/__tests__/scss/__snapshots__/styles-test.js.snap (hunk 1 above)
Comment (SimonFinney) [reply src/__tests__/scss/__snapshots__/styles-test.js.snap:3597]: There is! This is a snapshot of the changes to the CSS output.

**Review by areddon — APPROVED** [review]

**Pull request merged**